creator-node/src/services/stateMachineManager/stateReconciliation/issueSyncRequest.jobProcessor.js

```diff
@@ -1,37 +1,264 @@
+const axios = require('axios')
+
+const config = require('../../../config')
+const models = require('../../../models')
+const Utils = require('../../../utils')
+const { retrieveClockValueForUserFromReplica } = require('../stateMachineUtils')
+const { enqueueSyncRequest } = require('./stateReconciliationUtils')
+const SyncRequestDeDuplicator = require('./SyncRequestDeDuplicator')
+const SecondarySyncHealthTracker = require('./SecondarySyncHealthTracker')
+const { SYNC_MONITORING_RETRY_DELAY_MS } = require('../stateMachineConstants')
+
+const thisContentNodeEndpoint = config.get('creatorNodeEndpoint')
+const secondaryUserSyncDailyFailureCountThreshold = config.get(
+  'secondaryUserSyncDailyFailureCountThreshold'
+)
+const maxSyncMonitoringDurationInMs = config.get(
+  'maxSyncMonitoringDurationInMs'
+)
+
 /**
- * Processes a job to issue a sync request for a user.
- * Only issues request to a healthy secondary when this node is the user's primary and has a clock value
- * greater than the secondary's clock value.
+ * Processes a job to issue a request to perform a manual or recurring sync (determined by syncType param).
+ * The sync request syncs a user's data from this node (the user's primary)
+ * to another node (one of the user's secondaries).
  *
- * @param {number} jobId the id of the job being run
- * @param {Object[]} users users to find mismatched clock values for and, as necessary, issue requests to sync data from this node to the user's secondary/secondaries
- * @param {Set<string>} unhealthyPeers the set of unhealthy content nodes that one or more `users` has as their primary or secondary
- * @param {Object} userSecondarySyncMetrics mapping of each secondary node to the success metrics the user has had syncing to it
- * @param {Object} replicaSetNodesToUserClockStatusesMap map(replica set node => map(userWallet => clockValue))
- * @returns {Object} data about which sync requests were successfully enqueued and which failed
+ * @param {Object} param job data
+ * @param {Object} param.logger the logger that can be filtered by jobName and jobId
+ * @param {string} param.syncType the type of sync (manual or recurring)
+ * @param {Object} param.syncRequestParameters axios params to make the sync request. Shape: { baseURL, url, method, data }
  */
-module.exports = async function (
-  jobId,
-  users, // TODO: This will change to be for a single user
-  unhealthyPeers,
-  userSecondarySyncMetricsMap,
-  replicaSetNodesToUserClockStatusesMap
-) {
-  // TODO: Extract the part of snapback's _aggregateOps() code that
-  //       finds potentialSyncRequests *into the monitoring queue*, and then
-  //       pass it here (update params here) so that this can copy snapback's `issueSyncRequestsToSecondaries()`
+module.exports = async function ({ logger, syncType, syncRequestParameters }) {
```

Comment:
this function is very similar to snapback's [processSyncOperation()](https://github.com/AudiusProject/audius-protocol/blob/master/creator-node/src/snapbackSM/snapbackSM.js#L1741)